Description: don't override an existing release agent
 If someone else has already installed a release agent, do not
 override it with our own.
 Without this, if libpam-cgfs is installed to run after pam-systemd,
 then user sessions are stopped as soon as libpam-cgfs moves the
 user session to a new cgroup.
Author: Serge Hallyn <serge.hallyn@ubuntu.com>

Index: systemd-shim-9/src/cgmanager.c
===================================================================
--- systemd-shim-9.orig/src/cgmanager.c
+++ systemd-shim-9/src/cgmanager.c
@@ -163,8 +163,23 @@ cgmanager_remove (const gchar *path)
 void
 cgmanager_move_self (void)
 {
+  GVariant *reply;
+  gchar *str;
+
   cgmanager_call ("MovePidAbs", g_variant_new ("(ssi)", "all", "/", getpid ()), G_VARIANT_TYPE_UNIT, NULL);
 
+  int need_agent = 1;
+  if (cgmanager_call ("GetValue", g_variant_new ("(sss)", "systemd", "/", "release_agent"), G_VARIANT_TYPE ("(s)"), &reply))
+  {
+          g_variant_get(reply, "(s)", &str);
+          g_variant_unref(reply);
+          need_agent = strlen(str) < 1;
+          g_free(str);
+  }
+
+  if (!need_agent)
+          return;
+
   /* install our systemd cgroup release handler */
   g_debug ("Installing cgroup release handler " LIBEXECDIR "/systemd-shim-cgroup-release-agent");
   cgmanager_call ("SetValue",
